pkg/controller/mizar/mizar-endpoints-controller.go

```diff
@@ -0,0 +1,372 @@
+limitations under the License.
+
+Reference:
+(1) https://github.com/futurewei-cloud/arktos.git, arktos/pkg/controller/endpoints_controller
+*/
+
+package mizar
+
+import (
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+const (
+	EndpointsKind          string = "Endpoints"
+	Endpoints_Ready        string = "True"
+	EndpointsStatusMessage string = "HANDLED"
+	EndpointsNoChange      int    = 1
+	EndpointsUpdate        int    = 2
+	EndpointsResume        int    = 3
+)
+
+type ServiceEndpoint struct {
+	name      string
+	namespace string
+	tenant    string
+	addresses []string
+	ports     []Ports
+}
+
+//frontPort: service' port, backendPort: endpoint' port
+//protocol: network protocol TCP by default
+type Ports struct {
+	frontPort   string
+	backendPort string
+	protocol    string
+}
+
+type MizarEndpointsController struct {
+	kubeclientset       *kubernetes.Clientset
+	informer            coreinformers.EndpointsInformer
+	informerSynced      cache.InformerSynced
+	lister              corelisters.EndpointsLister
+	serviceListerSynced cache.InformerSynced
+	serviceLister       corelisters.ServiceLister
+	recorder            record.EventRecorder
+	queue               workqueue.RateLimitingInterface
+	grpcHost            string
+}
+
+func NewMizarEndpointsController(kubeclientset *kubernetes.Clientset, endpointInformer coreinformers.EndpointsInformer, serviceInformer coreinformers.ServiceInformer, grpcHost string) (*MizarEndpointsController, error) {
+	informer := endpointInformer
+	eventBroadcaster := record.NewBroadcaster()
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "mizar-endpoints-controller"})
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(
+		&v1core.EventSinkImpl{Interface: kubeclientset.CoreV1().EventsWithMultiTenancy(metav1.NamespaceAll, metav1.TenantAll)})
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	c := &MizarEndpointsController{
+		kubeclientset:       kubeclientset,
+		informer:            informer,
+		informerSynced:      informer.Informer().HasSynced,
+		lister:              informer.Lister(),
+		serviceListerSynced: serviceInformer.Informer().HasSynced,
+		serviceLister:       serviceInformer.Lister(),
+		recorder:            recorder,
+		queue:               queue,
+		grpcHost:            grpcHost,
+	}
+	klog.Infof("Sending events to api server")
```

Comment:
This log does not provide any information, also need to use Info instead of Infof